linux_cross.nix

```diff
@@ -0,0 +1,69 @@
+let
+  # Here we try to figure out which qemu to use based on the host platform.
+  # This guess can be overriden by passing qemuSuffix
+  qemuByHostPlatform = hostPlatform:
+    # I'd prefer this was a dictionary lookup, with a fall through into abort,
+    # that would make this more readable I guess.  I think there is some similar
+    # mapping somewhere in haskell.nix
+    if hostPlatform.isAarch32
+    then "arm"
+    else if hostPlatform.isAarch64
+    then "aarch64"
+    else abort "Don't know which QEMU to use for hostPlatform ${hostPlatform.config}. Please provide qemuSuffix";
+in
+{ stdenv
+, lib
+, writeScriptBin
+, qemu
+, qemuSuffix ? (qemuByHostPlatform hostPlatform)
+, iserv-proxy
+, remote-iserv
```

Comment:
I'm not sure if you even need remote-iserv here? You should be able to just wrap `${qemu}/bin/qemu-${qemuSuffix} ${ghc}/bin/ghc-iserv``` and put it in ```-pgmi```. Haven't tried but might make some of this logic simpler.

Reply:
Perhaps @angerman knows if there is some way around the complicated iserv logic? It doesn't look like you can just put ghci in ```-pgmi```

Reply:
@matthewbauer, you’ll just need to wrap off into a shell script. `-pgmi` expects a single callable. `writeShellScriptBin` with
```
qemu-arm iserv $@
```

Is probably enough. Brevity due to typing this on a phone.

the benefit that iserv-proxy provides is that you can get a better glimpse at what’s going on if at the message level if you pass `-v`.